driver/src/orderbook/shadow_orderbook.rs

```diff
@@ -30,15 +30,15 @@ impl<'a> ShadowedOrderbookReader<'a> {
     /// Create a new instance of a shadowed orderbook reader that starts a
     /// background thread
     pub fn new(
-        primary: &'a dyn StableXOrderBookReading,
-        shadow: Box<dyn StableXOrderBookReading + Send + 'static>,
+        primary: &'a (dyn StableXOrderBookReading + Sync),
+        shadow: impl StableXOrderBookReading + Send + 'static,
```

Comment:
Would it be semantically simpler of both of these were passed in the same way (with ownership)?

Reply:
I think the reference is fine. The only bit that was a bit of a problem is the `shadow` instance since it is run on a separate thread and so it either needs to be `'static` which is what we use here (since the type is owned) or use a scoped thread, then the shadow needs to live as long as the scope. So that means that scope needs to live in `main.rs` and leaks a bit of the implementation details.